server/src/test/java/io/druid/segment/realtime/plumber/RealtimePlumberSchoolTest.java

```diff
@@ -297,4 +319,144 @@ public void run()
 
     Assert.assertEquals(1, metrics.failedPersists());
   }
+
+  @Test(timeout = 60000)
+  public void testPersistHydrantGaps() throws Exception
+  {
+    final Object commitMetadata = "dummyCommitMetadata";
+    testPersistHydrantGapsHelper(commitMetadata);
+  }
+
+  private void testPersistHydrantGapsHelper(final Object commitMetadata) throws Exception
+  {
+    final MutableBoolean committed = new MutableBoolean(false);
+    Interval testInterval = new Interval(new DateTime("1970-01-01"), new DateTime("1970-12-31"));
+
+    RealtimePlumber plumber2 = (RealtimePlumber) realtimePlumberSchool.findPlumber(schema2, tuningConfig, metrics);
+    plumber2.getSinks()
+           .put(
+               0L,
+               new Sink(
+                   testInterval,
+                   schema2,
+                   tuningConfig,
+                   new DateTime("2014-12-01T12:34:56.789").toString()
+               )
+           );
+    Assert.assertNull(plumber2.startJob());
+
+    final Committer committer = new Committer()
+    {
+      @Override
+      public Object getMetadata()
+      {
+        return commitMetadata;
+      }
+
+      @Override
+      public void run()
+      {
+        committed.setValue(true);
+      }
+    };
+    plumber2.add(getTestInputRow("1970-01-01"), Suppliers.ofInstance(committer));
+    plumber2.add(getTestInputRow("1970-02-01"), Suppliers.ofInstance(committer));
+    plumber2.add(getTestInputRow("1970-03-01"), Suppliers.ofInstance(committer));
+    plumber2.add(getTestInputRow("1970-04-01"), Suppliers.ofInstance(committer));
+    plumber2.add(getTestInputRow("1970-05-01"), Suppliers.ofInstance(committer));
+
+    plumber2.persist(committer);
+
+    while (!committed.booleanValue()) {
+      Thread.sleep(100);
+    }
+    plumber2.getSinks().clear();
+    plumber2.finishJob();
+
+    File persistDir = plumber2.computePersistDir(schema2, testInterval);
+
+    /* Create some gaps in the persisted hydrants and reload */
```

Comment:
I think it'd be nice to have an assert here about which persist directories should have been written.